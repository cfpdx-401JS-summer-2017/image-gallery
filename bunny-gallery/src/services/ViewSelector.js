import React, { Component } from 'react';
import ImageList from '../views/ImageList';
import ImageThumbnail from '../views/ImageThumbnail';
import ImageGallery from '../views/ImageGallery';

const View = {
    list: ImageList,
    thumbnail: ImageThumbnail,
    gallery: ImageGallery
}

const views = Object.keys(View);

class ViewSelector extends Component {

    constructor(props) {
        super(props);
        this.state = {
            view: views[1],
            views: views,
        }
    }

    render() {
        const { images, onRemove } = this.props;
        const { views, view } = this.state;
        const ImageView = View[view];


        return (
            <div className="App">
                <div>
                    {views.map(v => (
                        <button key={v} onClick={() => this.setState({ view: v })}>
                            {v}</button>))}
                    <ImageView images={images} views={views} onRemove = {onRemove}/>
                </div>
            </div>
        );
    }
}

export default ViewSelector;