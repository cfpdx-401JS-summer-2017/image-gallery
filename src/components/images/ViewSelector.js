import React, { Component } from 'react';
import ImageList from '../views/ImageList';
import ImageThumbnail from '../views/ImageThumbnail';
import ImageGallery from '../views/ImageGallery';
import { Switch, Route, Link, withRouter } from 'react-router-dom';
import qs from 'qs';

export const View = {
    list: ImageList,
    thumbnail: ImageThumbnail,
    gallery: ImageGallery
}

export const views = Object.keys(View);

function ViewSelector(props) {

    const { images, onRemove, match, location, history } = props;
    const view = qs.parse(location.search.slice(1)).view || 'thumbnail';
    const ImageView = View[view];



    return (
        <div className="App">
            <div>
                {views.map(v => (
                    <button key={v} disabled={v === view} onClick={() => history.push({ search: `?view=${v}` })}>
                        {v}</button>))}
                <ImageView images={images} onRemove={onRemove} />
            </div>
        </div>
    );
}

export default withRouter(ViewSelector);