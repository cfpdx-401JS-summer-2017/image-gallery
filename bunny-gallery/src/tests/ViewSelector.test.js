
import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import ViewSelector from '../services/ViewSelector'

describe('ViewSelector', () => {

    it('switches to correct view', () => {
        const wrapper = shallow(<ViewSelector images={[]} />);
        expect(toJSON(wrapper)).toMatchSnapshot();

        const component = wrapper.instance();
        
        component.handleChange( 'gallery' );
        wrapper.update();
        expect(toJSON(wrapper)).toMatchSnapshot();

        component.handleChange( 'thumbnail' );
        wrapper.update();
        expect(toJSON(wrapper)).toMatchSnapshot();

        component.handleChange( 'list' );
        wrapper.update();
        expect(toJSON(wrapper)).toMatchSnapshot();

    })
})