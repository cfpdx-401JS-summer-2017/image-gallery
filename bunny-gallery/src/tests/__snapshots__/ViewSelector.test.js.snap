// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewSelector switches to correct view 1`] = `
<div
  className="App"
>
  <div>
    <button
      onClick={[Function]}
    >
      list
    </button>
    <button
      onClick={[Function]}
    >
      thumbnail
    </button>
    <button
      onClick={[Function]}
    >
      gallery
    </button>
    <ImageThumbnail
      images={Array []}
      views={
        Array [
          "list",
          "thumbnail",
          "gallery",
        ]
      }
    />
  </div>
</div>
`;

exports[`ViewSelector switches to correct view 2`] = `
<div
  className="App"
>
  <div>
    <button
      onClick={[Function]}
    >
      list
    </button>
    <button
      onClick={[Function]}
    >
      thumbnail
    </button>
    <button
      onClick={[Function]}
    >
      gallery
    </button>
    <ImageGallery
      images={Array []}
      views={
        Array [
          "list",
          "thumbnail",
          "gallery",
        ]
      }
    />
  </div>
</div>
`;

exports[`ViewSelector switches to correct view 3`] = `
<div
  className="App"
>
  <div>
    <button
      onClick={[Function]}
    >
      list
    </button>
    <button
      onClick={[Function]}
    >
      thumbnail
    </button>
    <button
      onClick={[Function]}
    >
      gallery
    </button>
    <ImageThumbnail
      images={Array []}
      views={
        Array [
          "list",
          "thumbnail",
          "gallery",
        ]
      }
    />
  </div>
</div>
`;

exports[`ViewSelector switches to correct view 4`] = `
<div
  className="App"
>
  <div>
    <button
      onClick={[Function]}
    >
      list
    </button>
    <button
      onClick={[Function]}
    >
      thumbnail
    </button>
    <button
      onClick={[Function]}
    >
      gallery
    </button>
    <ImageList
      images={Array []}
      views={
        Array [
          "list",
          "thumbnail",
          "gallery",
        ]
      }
    />
  </div>
</div>
`;
